### ISSUE TITLE
Create an own Python package for the REST interface

### ISSUE THREAD

**Issue by FlorianWilhelm** [Yelp/MOE#461]
Currently the installation of MOE is quite hard for beginners. Although a docker image is provided which makes is easy to get it running, in order to use the provided Python REST interface one still has to install MOE somehow locally or start messing around with the PYTHONPATH variable which is just nasty.
My suggestion would be to split the whole Python REST interface into an own library called ``moe_api`` for instance. The installation for dummies would then be as easy as running the provided docker image and typing ``pip install moe_api`` in a virtuelenv or conda environment. Since ``moe_api`` would be plain Python code there is nothing to compile and even a port of that client to Python 3 would be so much easier. During this cleanup one could surely simplify the current interface.

**Comment by mulyoved** [comment]
Forked MOE and striped some code so now have the REST client with only minimal dependencies and no C++ code

Only dependancies are:
- simplejson
- numpy
- colander

https://github.com/mulyoved/clientMOE

It is only quick and totally dirty hack for now

**Comment by FlorianWilhelm** [comment]
@mulyoved Yes, great! Looking forward to use it. Will you upload a package to PyPI?

**Comment by mulyoved** [comment]
published as PyPI package 

`pip install clientMOE`

**Comment by FlorianWilhelm** [comment]
@mulyoved Nice, just installed it with pip! Thanks a bunch! 
But what about a cooler name? Maybe SCHMOE, the Small & Concise Helper for MOE ;-) At least the package name itself which is ``moe`` right now should be renamed so that there is no ambiguity with MOE itself if anyone installs both packages.

**Comment by RokoMijic** [comment]
My understanding is that clientMOE requires python 2.7. What are the barriers to making it python 3 compatible?

**Comment by RokoMijic** [comment]
I have forked this and done a quick-and-dirty conversion to python3 (3.5.2). 

[https://github.com/RokoMijic/clientMOE](https://github.com/RokoMijic/clientMOE)

Due to @mulyoved 's code not including the CPP wrappers, a number of tests simply had to be deleted since they relied on that code. 

There are 91 remaining tests which all pass in python 3.5.2.

**Comment by ecashin** [comment]
If there was API documentation in the pretty endpoints, not just examples, then in Python it would be easy to use the `requests` module as a client.  The Python 2/3 split would not be a problem, and you could use any language you want, not just Python.

The API documentation would specify the JSON schema for messages, using something like Orderly, which documents constraints and formats for the messages, not specific examples, which leave the abstraction implicit and potentially ambiguous.
